### PR TITLE
Update renovate gh action

### PR DESCRIPTION
**Pull request by vicentepinto98**

### CODE REVIEW

File: .github/workflows/renovate.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
       - name: Self-hosted Renovate
-        uses: renovatebot/github-action@v39.2.4
+        uses: renovatebot/github-action@v40
```
Comment (hcsa73) [inline .github/workflows/renovate.yaml:16]: It doesn't require a minor and patch version? I thought it did

Comment (vicentepinto98) [reply .github/workflows/renovate.yaml:16]: I dont think so, the action above also has just major version

Comment (hcsa73) [reply .github/workflows/renovate.yaml:16]: That's not enough, that depends on what each GitHub action has released, and different actions have different releases
Also it doesn't work: https://github.com/stackitcloud/stackit-sdk-go/actions/runs/7744970877/job/21119926497

File: .github/workflows/renovate.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
       - name: Self-hosted Renovate
-        uses: renovatebot/github-action@v39.2.4
+        uses: renovatebot/github-action@v40.0.0
```
Comment (hcsa73) [inline .github/workflows/renovate.yaml:16]:
```suggestion
        uses: renovatebot/github-action@v40.0.2
```
https://github.com/renovatebot/github-action/releases

**Review by hcsa73 — APPROVED** [review]

**Pull request merged**